services/src/main/resources/org/keycloak/userprofile/config/keycloak-default-user-profile.json

```diff
@@ -52,5 +52,12 @@
 				"person-name-prohibited-characters": {}
 			}
 		}
+	],
+	"groups": [
+		{
+			"name": "user-metadata",
+			"displayHeader": "User metadata",
```

Comment:
I'm not sure what is the best name for this group too. But `User metadata` seems too generic and perhaps `LDAP` or `LDAP Metadata` or `LDAP Internal Attributes` or something else along these lines?

Or are you thinking about adding not only LDAP attributes to this group? Like any other internal-use attribute?

Reply:
Yes, I was thinking about this as `User metadata`, so it is not only specific to LDAP, but it is possible to add here any other metadata attributes. I am planning to update this PR to handle also `Kerberos provider` and `SSSD provider` and I expect to use same group `User metadata` for those providers as well. Also administrator can possibly use this group to add some attributes manually though? Maybe the attributes like `userCertificate` we discussed before or some others?

Reply:
Works for me. But I'm not sure if having separate groups has a better UX though. We can always improve this later and ask UI team for feedback too.

Reply:
@ssilvert @jonkoops @edewit What do you think about this?

In summary: This PR introduces default group "user metadata" into `keycloak-default-user-profile.json`. It is aimed to be used for "metadata attributes" related to user-profile. Not for normal attributes like "firstName", "lastname", "birthday" etc. Examples used in relation to LDAP storage provider are attributes like `LDAP_ID` or `LDAP_ENTRY_DN`. See screenshot above.

Does it makes sense to have separate metadata group like this or should we go with the "flat" structure and treat these attributes as normal attributes?

Reply:
I think having these in a group makes sense.  

We may want to talk about how much control the administrator has over the attribute settings.  For instance, what happens if an admin deletes the attribute from User Profile?

Reply:
@ssilvert Administrator has possibility to override the settings for the particular auto-generated attribute as when he define it in user-profile, then the defined version has preference over the auto-generated. This is done here https://github.com/keycloak/keycloak/pull/24282/files#diff-bb4ffa08fae23dc22710d174b81fadf5028b58a8742d9fba4035a2c916e92982R1014 .

So for instance if administrator doesn't want `LDAP_ID` to be shown in the attributes, he can define the `LDAP_ID` attribute in user-profile and disable all permissions. 

The assumption is that most of the administrators will be OK with default settings of "metadata attributes" like LDAP_ID, but there is still possibility to override the default by define it in user profile. 

Is it what you meant?